src/runner/settings_window.cpp

```diff
@@ -206,6 +206,37 @@ BOOL run_settings_non_elevated(LPCWSTR executable_path, LPWSTR executable_args,
     return process_created;
 }
 
+// This variable is set based on the os version
+bool use_old_settings = true;
+const int RS5_VERSION = 17999;
+
+// This function obtains the os build number from the registry and accordingly sets the settings executable
+void os_detection()
+{
+    HKEY hkey;
+    const wchar_t* osVersionPath = L"SOFTWARE\\Microsoft\\Windows NT\\CurrentVersion";
+    
+    if (RegOpenKeyExW(HKEY_LOCAL_MACHINE, osVersionPath, 0, KEY_READ, &hkey) == ERROR_SUCCESS)
+    {
+        wchar_t buildNumberString[256];
```

Comment:
⚠ 256 is not a safe assumption here. Instead, call RegQueryValueEx twice, once with a nullptr buffer, trap on ERROR_MORE_DATA, then build an appropriate buffer and call again. (Docs have an example of [this common pattern](https://docs.microsoft.com/en-us/windows/win32/api/winreg/nf-winreg-regqueryvalueexw#remarks).)